src/layouts/EditHomepage/utils.ts

```diff
@@ -1,7 +1,72 @@
 import _ from "lodash"
 
-export const getErrorValues = (
+import {
+  EditorHomepageState,
+  HeroFrontmatterSection,
+  isDropdownSection,
+  isHighlightSection,
+} from "types/homepage"
+import { validateDropdownElem, validateHighlight, validateSection } from "utils"
+
+export const getDefaultValues = (
   obj: Record<string, string>
 ): Record<string, string> => {
   return _.mapValues(obj, () => "")
 }
+
+// NOTE: We might wish to extend this so that we map the values to
+// error msg etc then extract an overall boolean based on that
+// but for now will return a bool
+export const getHasErrorFromHomepageState = ({
+  frontMatter,
+}: EditorHomepageState) => {
+  // we validate the frontmatter to check that all the fields are valid
+  const { sections } = frontMatter
+  const [initialSection, ...rest] = sections
+  const heroSection = (initialSection as HeroFrontmatterSection).hero
+
+  let hasError = false
+  if (isHighlightSection(heroSection)) {
+    hasError = heroSection.key_highlights.reduce((acc, cur) => {
+      const curErr = _(cur)
+        .entries()
+        .map(([key, value]) => validateHighlight(key, value))
+        .some()
+        .valueOf()
+      return acc || curErr
+    }, hasError)
+  }
+
```

Comment:
Looking at the logic, if `hasError` is `true` at this point, we can directly return `true`